src/main/java/com/minecolonies/entity/ai/citizen/lumberjack/Tree.java

```diff
@@ -105,18 +115,23 @@ else if(block instanceof BlockNewLog)
             }
 
             location = getBaseLog(world, log);
+            topLog = getTopLog(world, log);
+
             woodBlocks = new LinkedList<>();
+            addAndSearch(world, location);
+
             checkTree(world, getTopLog(world, log));
             stumpLocations = new ArrayList<>();
+            woodBlocks = new LinkedList<>();
```

Comment:
clear is better from an encapsulation perspective. We have to ask us: "Do we want LinkedList or ArrayList?" And with clear we don't have to decide this every time anew.
Performance wise .clear is faster as we already have a decent sized list (cleared in O(n)). But always use .clear and not .removeAll